rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -3172,24 +3275,33 @@ namespace rsx
 		m_profiler.enabled = !!g_cfg.video.overlay;
 	}
 
-	void thread::request_emu_flip(u32 buffer)
+	bool thread::request_emu_flip(u32 buffer)
 	{
 		if (is_current_thread()) // requested through command buffer
 		{
 			// NOTE: The flip will clear any queued flip requests
 			handle_emu_flip(buffer);
+			return true;
 		}
 		else // requested 'manually' through ppu syscall
 		{
 			if (async_flip_requested & flip_request::emu_requested)
 			{
 				// ignore multiple requests until previous happens
-				return;
+				return true;
+			}
+
+			if (state & cpu_flag::exit)
```

Comment:
This feels off. Shouldn't this be backwards? RSX should be the last one to go down being a hardware unit, so why would a PPU thread get here? Missing check_state() somewhere? I'm a bit unsure on this one.

Reply:
No it's because RSX could have ignored the flip.